scripts/src/custard.ts

```diff
@@ -0,0 +1,530 @@
+  return errors;
+}
+
+function check(
+  kvs: any,
+  key: string,
+  isType: (x: any) => boolean,
+  err: string,
+): string[] {
+  // Fields are not required by default.
+  // Required fields must be checked explicitly.
+  const k = key.split('.').pop() || key;
+  if (kvs && kvs[k] && !isType(kvs[k])) {
+    return [`'${key}' must be ${err}, got: ${JSON.stringify(kvs[k])}`];
+  }
+  return [];
+}
+
+function checkString(kvs: any, key: string): string[] {
+  return check(kvs, key, isString, 'string');
+}
+
+function checkStringOrStrings(kvs: any, key: string): string[] {
+  return check(kvs, key, isStringOrStrings, 'string or string[]');
+}
+
+function checkDefinitions(kvs: any, key: string): string[] {
+  return check(kvs, key, isMapStringString, '{string: string} definitions');
+}
+
+function isString(x: any): boolean {
+  return typeof x === 'string';
+}
+
+function isArray(xs: any, isElem: (x: any) => boolean): boolean {
+  if (!Array.isArray(xs)) {
+    return false;
+  }
+  for (const x of xs) {
+    if (!isElem(x)) {
+      return false;
+    }
+  }
+  return true;
+}
+
+function isStringOrStrings(x: any): boolean {
+  return isString(x) || isArray(x, isString);
+}
+
+function isMapStringString(kvs: any): boolean {
+  if (typeof kvs !== 'object') {
+    return false;
+  }
+  for (const key in kvs) {
+    if (typeof key !== 'string' || typeof kvs[key] !== 'string') {
+      return false;
+    }
+  }
+  return true;
+}
+/* eslint-enable  @typescript-eslint/no-explicit-any */
+
+switch (process.argv[2]) {
+  case 'lint': {
+    const usageLint = usage('lint <config-path> [package-path...]');
+    const configPath = process.argv[3];
+    if (!configPath) {
+      console.error('Please provide the config file path.');
+      throw new Error(usageLint);
+    }
+    const packagePaths = process.argv.slice(4);
+    if (packagePaths.length === 0) {
+      console.error('Please provide the paths to lint.');
+      throw new Error(usageLint);
+    }
+    lint(configPath, packagePaths);
+    break;
+  }
+
+  case 'test': {
+    const usageTest = usage('test <config-path> <package-path>');
+    const configPath = process.argv[3];
+    if (!configPath) {
+      console.error('Please provide the config file path.');
+      throw new Error(usageTest);
+    }
+    const packagePaths = process.argv.slice(4);
+    if (packagePaths.length === 0) {
+      console.error('Please provide the paths to test.');
+      throw new Error(usageTest);
+    }
+    test(configPath, packagePaths);
+    break;
+  }
+
+  case undefined: {
+    // If no command was passed, just show the usage without an error.
+    console.log(usage('[lint | test] [options]'));
+    break;
+  }
+
+  default: {
+    // Only throw an error if running the script directly.
+    // Otherwise, this file is being imported (for example, on tests).
+    if (process.argv[1] && process.argv[1].match(/custard\.(ts|js)$|^-$/)) {
+      throw new Error(usage('[lint | test] [options]'));
```

Comment:
I get this warning when running under node v23:

```
> node --trace-warnings src/custard.ts
usage: node custard.ts [lint | test] [options]
(node:78683) ExperimentalWarning: Type Stripping is an experimental feature and might change at any time
    at emitExperimentalWarning (node:internal/util:285:11)
    at stripTypeScriptModuleTypes (node:internal/modules/typescript:166:5)
    at ModuleLoader.<anonymous> (node:internal/modules/esm/translators:551:16)
    at #translate (node:internal/modules/esm/loader:534:12)
    at ModuleLoader.loadAndTranslate (node:internal/modules/esm/loader:581:27)
    at async ModuleJob._link (node:internal/modules/esm/module_job:116:19)
```

Reply:
Yes, that warning still shows up in Node 23. From looking at the history, Node 22 introduced the feature under an experimental flag, then 23 removed the experimental flag and made it the default, but looks like the warning is still there. It's not a big issue and it still works as expected.

If we don't want that warning to show up, we can have a built `.js` file and a check that makes sure the js file is up to date. Later on we could have the js file as part of the release published artifacts, but that seems to be a bit trickier to fetch from workflows, so having the js file in git might be the simplest option.

Reply:
This warning can be disabled with `export NODE_OPTIONS="--no-warnings=75058"`